dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/UserDataFieldFillerTest.java

```diff
@@ -0,0 +1,254 @@
+package edu.harvard.iq.dataverse.dataset;
+
+import com.google.common.collect.Lists;
+import edu.harvard.iq.dataverse.DatasetFieldServiceBean;
+import edu.harvard.iq.dataverse.common.DatasetFieldConstant;
+import edu.harvard.iq.dataverse.persistence.dataset.ControlledVocabularyValue;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetField;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldCompoundValue;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldType;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.dataset.FieldType;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUserLookup;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+public class UserDataFieldFillerTest {
+
+    @InjectMocks
+    private UserDataFieldFiller fieldFiller;
+    
+    @Mock
+    private DatasetFieldServiceBean fieldService;
+    
+    private AuthenticatedUser user;
+    
+    @BeforeEach
+    public void beforeEach() {
+        user = new AuthenticatedUser();
+        user.setFirstName("John");
+        user.setLastName("Doe");
+        user.setAffiliation("Aff");
+        user.setEmail("example@domain.com");
+        user.setAuthenticatedUserLookup(new AuthenticatedUserLookup("some_id", "provider"));
+                
+    }
+    
+    // -------------------- TESTS --------------------
+    
+    @Test
+    public void fillUserDataInDatasetFields__FILL_DEPOSITOR() {
```

Comment:
In cases where we use the same method we can utilize Junit5 `@ParameterizedTest` with `@MethodSource`, for me it's a minor thing so just a suggestion.

Reply:
Parametrized test here is kind of troublesome because asserts depends on type of field primitive/compund.

Reply:
Ah true, it would have to be 2 parametrized tests.